extensions/azurecore/src/utils/Logger.ts

```diff
@@ -6,19 +6,52 @@
 import * as vscode from 'vscode';
 import * as loc from '../localizedConstants';
 
+export enum LogLevel {
+	'All',
+	'Off',
+	'Critical',
+	'Error',
+	'Warning',
+	'Information',
+	'Verbose',
+}
+
 export class Logger {
 
-	public static channel: vscode.OutputChannel = vscode.window.createOutputChannel(loc.extensionName);
 
+	public static channel: vscode.OutputChannel = vscode.window.createOutputChannel(loc.extensionName);
 	private static _piiLogging: boolean = false;
+	public static config = vscode.workspace.getConfiguration('mssql');
 
-	static log(msg: any, ...vals: any[]) {
-		const fullMessage = `${msg} - ${vals.map(v => JSON.stringify(v)).join(' - ')}`;
-		this.channel.appendLine(fullMessage);
+	static write(logLevel: LogLevel, msg: any, ...vals: any[]) {
+		switch (logLevel) {
+			case LogLevel.Error:
+				if (LogLevel[logLevel] === this.config.tracingLevel || this.config.tracingLevel === 'Verbose') {
```

Comment:
These checks don't appear to be handling other more verbose logging levels - for example if the logging level is All, Warning, Information or Verbose I would expect Error events to be logged.